tests/system/python/api/test_notification.py

```diff
@@ -0,0 +1,239 @@
+          "notification_type": "one shot"}, '400: Invalid rule property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1], "channel": ":",
+          "enabled": True, "notification_type": "one shot"}, '400: Invalid channel property in payload.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": NOTIFY_INBUILT_RULES[1],
+          "channel": NOTIFY_PLUGIN, "enabled": "bla", "notification_type": "one shot"},
+         '400: Only "true", "false", true, false are allowed for value of enabled.'),
+        ({"name": "Test4", "description": "Test4_Notification", "rule": "InvalidRulePlugin",
+          "channel": "InvalidChannelPlugin", "enabled": True, "notification_type": "one shot"},
+         '400: Invalid rule plugin InvalidRulePlugin and/or delivery plugin InvalidChannelPlugin supplied.')
+    ])
+    def test_invalid_create_notification_instance(self, foglamp_url, test_input, expected_error):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/notification', json.dumps(test_input))
+        r = conn.getresponse()
+        assert 400 == r.status
+        r = r.read().decode()
+        assert expected_error == r
+
+    def test_create_valid_notification_instance(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/notification', json.dumps(DATA))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "Notification {} created successfully".format(DATA['name']) == jdoc['result']
+
+        conn.request("GET", '/foglamp/notification')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        actual_data = jdoc['notifications'][0]
+        assert DATA['name'] == actual_data['name']
+        assert DATA['channel'] == actual_data['channel']
+        assert 'true' == actual_data['enable']
+        assert DATA['notification_type'] == actual_data['notificationType']
+        assert DATA['rule'] == actual_data['rule']
+
+        conn.request("GET", '/foglamp/notification/plugin')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert 2 == len(jdoc)
+        assert NOTIFY_PLUGIN == jdoc['delivery'][0]['name']
+        assert "notificationDelivery" == jdoc['delivery'][0]['type']
+        assert 2 == len(jdoc['rules'])
+
+    @pytest.mark.parametrize("test_input, expected_error", [
+        pytest.param({"name": "Test8"}, '400: Name update is not allowed.', marks=pytest.mark.skip(reason="FOGL-2673")),
+        pytest.param({"name": "="}, '400: Invalid name property in payload.', marks=pytest.mark.skip(reason="FOGL-2673")),
```

Comment:
better to mark xfail. 
Nice to learn the way to mark skip/xfail in parameterized vector.